cmd/kuberhealthy/kuberhealthy.go

```diff
@@ -1292,20 +1291,46 @@ func (k *Kuberhealthy) externalCheckReportHandler(w http.ResponseWriter, r *http
 	// make a request ID for tracking this request
 	requestID := "web: " + uuid.New().String()
 
-	k.externalCheckReportHandlerLog(requestID, "Client connected to check report handler from", r.RemoteAddr, r.UserAgent())
+	k.externalCheckReportHandlerLog(requestID, "Client connected to check report handler from", r.UserAgent())
 
-	// validate the calling pod to ensure that it has a proper KH_CHECK_NAME and KH_RUN_UUID
-	k.externalCheckReportHandlerLog(requestID, "validating external check status report from: ", r.RemoteAddr)
-	ipReport, err := k.validateExternalRequest(r.RemoteAddr)
-	if err != nil {
-		w.WriteHeader(http.StatusBadRequest)
-		k.externalCheckReportHandlerLog(requestID, "Failed to look up pod by IP:", r.RemoteAddr, err)
-		return nil
+	// Get kh-run-uuid from the request header and verify it exists. If the kh-run-uuid doesn't exist, verify request by
+	// validating using the pod's remote IP.
+	reportingKHRunUUID := r.Header.Get("kh-run-uuid")
+	var podReport PodReportInfo
+	var err error
+	if len(reportingKHRunUUID) == 0 {
+		k.externalCheckReportHandlerLog(requestID, "Received blank kh-run-uuid header from external request. Validating external check status report from the pod's remote IP.")
+
+		// break the port off the RemoteAddr incoming string
+		ipPortString := strings.Split(r.RemoteAddr, ":")
+		if len(ipPortString) == 0 {
```

Comment:
Whenever I see a `len()` check on a slice, I wonder if I should instead be fetching the slice from some function that is smart enough to return an error.  This way errors are explicit values, rather than ascertained by the length of the slice in your calling code.